stream/stream_lavf.c

```diff
@@ -238,6 +238,90 @@ void mp_setup_av_network_options(AVDictionary **dict, const char *target_fmt,
     talloc_free(temp);
 }
 
+#define ADD_PROTO(name, ...) \
+    else if (!strcmp(name, ffmpeg_protos[i])) { \
+        const char **protos = (const char *[]) {name, __VA_ARGS__}; \
+        for (int j = 0; protos[j]; j++) \
+            MP_TARRAY_APPEND(NULL, protocols, num, talloc_strdup(protocols, protos[j])); \
+    }
+
+static char **get_safe_protocols(void)
+{
+    int num = 0;
+    char **protocols = NULL;
+    char **ffmpeg_demuxers = mp_get_lavf_demuxers();
+    char **ffmpeg_protos = mp_get_lavf_protocols();
+
+    for (int i = 0; ffmpeg_protos[i]; i++) {
+        if (0) {}
+        ADD_PROTO("data", NULL)
+        ADD_PROTO("gopher", NULL)
+        ADD_PROTO("gophers", NULL)
+        ADD_PROTO("http", "dav", "webdav", NULL)
+        ADD_PROTO("httpproxy", NULL)
+        ADD_PROTO("https", "davs", "webdavs", NULL)
+        ADD_PROTO("ipfs", NULL)
+        ADD_PROTO("ipns", NULL)
+        ADD_PROTO("mmsh", "mms", "mmshttp", NULL)
+        ADD_PROTO("mmst", NULL)
+        ADD_PROTO("rist", NULL)
+        ADD_PROTO("rtmp", NULL)
+        ADD_PROTO("rtmpe", NULL)
+        ADD_PROTO("rtmps", NULL)
+        ADD_PROTO("rtmpt", NULL)
+        ADD_PROTO("rtmpte", NULL)
+        ADD_PROTO("rtmpts", NULL)
+        ADD_PROTO("rtp", NULL)
+        ADD_PROTO("srt", NULL)
+        ADD_PROTO("srtp", NULL)
```

Comment:
Maybe move those to static array and add loop for them?

Reply:
How would you handle the ones with multiple names though? Also kind of wanted to avoid a loop in a loop (not that it really matters but still).

Reply:
It can be an array of `(const char *[]) {name, __VA_ARGS__};` things. Don' worry about the loop, it is no different to current state, it is just manually unrolled.